huaweicloud/services/cci/resource_huaweicloud_cciv2_namespace.go

```diff
@@ -0,0 +1,288 @@
+	}
+	return nil
+}
+
+func resourceNamespaceRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.CciV2Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("Error creating CCI v2 client: %s", err)
+	}
+
+	resp, err := GetNamespaceDetail(client, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error getting the specifies namespace form server")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("api_version", utils.PathSearch("apiVersion", resp, nil)),
+		d.Set("kind", utils.PathSearch("kind", resp, nil)),
+		d.Set("name", utils.PathSearch("metadata.name", resp, nil)),
+		d.Set("annotations", utils.PathSearch("metadata.annotations", resp, nil)),
+		d.Set("labels", utils.PathSearch("metadata.labels", resp, nil)),
+		d.Set("creation_timestamp", utils.PathSearch("metadata.creationTimestamp", resp, nil)),
+		d.Set("resource_version", utils.PathSearch("metadata.resourceVersion", resp, nil)),
+		d.Set("uid", utils.PathSearch("metadata.uid", resp, nil)),
+		d.Set("finalizers", utils.PathSearch("spec.finalizers", resp, nil)),
+		d.Set("status", utils.PathSearch("status.phase", resp, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceNamespaceUpdate(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceNamespaceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.CciV2Client(conf.GetRegion(d))
+	namespace := d.Id()
+	if err != nil {
+		return diag.Errorf("Error creating CCI v2 client: %s", err)
+	}
+
+	deleteNamespaceHttpUrl := "apis/cci/v2/namespaces/{name}"
+	deleteNamespacePath := client.Endpoint + deleteNamespaceHttpUrl
+	deleteNamespacePath = strings.ReplaceAll(deleteNamespacePath, "{name}", namespace)
+	deleteNamespaceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	_, err = client.Request("DELETE", deleteNamespacePath, &deleteNamespaceOpt)
+	if err != nil {
+		return diag.Errorf("error deleting the specifies namespace (%s): %s", namespace, err)
+	}
+
+	err = waitForDeleteNamespaceStatus(ctx, client, namespace, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func waitForDeleteNamespaceStatus(ctx context.Context, client *golangsdk.ServiceClient, ns string, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Active", "Terminating"},
+		Target:  []string{"DELETED"},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := GetNamespaceDetail(client, ns)
+			if err != nil {
+				if _, ok := err.(golangsdk.ErrDefault404); ok {
+					return resp, "DELETED", nil
```

Comment:
```suggestion
					return "", "DELETED", nil
```
if err is not nil, then resp is nil, so the first return param(resp) can not be nil